core/dbt/parser/generic_test_builders.py

```diff
@@ -257,7 +257,28 @@ def __init__(
             if not value and "config" in self.args:
                 value = self.args["config"].pop(key, None)
             if isinstance(value, str):
-                value = get_rendered(value, render_ctx, native=True)
+
+                try:
+                    value = get_rendered(value, render_ctx, native=True)
+                except UndefinedMacroException as e:
+
+                    # Generic tests do not include custom macros in the Jinja
+                    # rendering context, so this will almost always fail. As it
+                    # currently stands, the error message is inscrutable, which
+                    # has caused issues for some projects migrating from
+                    # pre-0.20.0 to post-0.20.0.
+                    # See https://github.com/dbt-labs/dbt-core/issues/4103
+                    # and https://github.com/dbt-labs/dbt-core/issues/5294
+                    raise_compiler_error(
+                        f"The {self.target.name}.{column_name} column's "
+                        f'"{self.name}" test references an undefined '
+                        f"macro in its {key} configuration argument. "
+                        f"The macro {e.msg}.\n"
+                        "Please note that the generic test configuration parser "
+                        "currently does not support using custom macros to "
+                        "populate configuration values"
```

Comment:
@jtcohen6 Do you have any thoughts in the wording of this exception?

Reply:
This new error message format looks great to me 👍 

The thing that will be super helpful is having all the additional context to help troubleshoot 💪